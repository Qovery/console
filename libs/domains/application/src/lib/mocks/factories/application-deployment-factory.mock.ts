import { DeploymentHistoryApplication, StateEnum } from 'qovery-typescript-axios'
import { Chance } from 'chance'

const chance = new Chance()

export const applicationDeploymentsFactoryMock = (howMany: number): DeploymentHistoryApplication[] =>
  Array.from({ length: howMany }).map((_, index) => ({
    id: index.toString(),
    created_at: new Date().toString(),
    updated_at: new Date().toString(),
    name: chance.name(),
    status: chance.pickone(Object.values([StateEnum.DEPLOYED, StateEnum.RUNNING, StateEnum.STOP_ERROR])),
  }))
