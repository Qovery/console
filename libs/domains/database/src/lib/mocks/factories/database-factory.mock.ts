import { Chance } from 'chance'
import { Database, DatabaseTypeEnum, DatabaseModeEnum, DatabaseAccessibilityEnum } from 'qovery-typescript-axios'

const chance = new Chance()

export const databaseFactoryMock = (howMany: number): Database[] =>
  Array.from({ length: howMany }).map((_, index) => ({
    id: `${index}`,
    created_at: new Date().toString(),
    updated_at: new Date().toString(),
    name: chance.name(),
    type: DatabaseTypeEnum.POSTGRESQL,
    version: '12',
    mode: DatabaseModeEnum.CONTAINER,
    accessibility: DatabaseAccessibilityEnum.PRIVATE,
    cpu: 1000,
    memory: 1024,
    storage: 1024,
    environment: {
      id: chance.guid(),
    },
    host: chance.name(),
    port: 80,
    maximum_cpu: 10,
    maximum_memory: 10,
    disk_encrypted: false,
  }))
