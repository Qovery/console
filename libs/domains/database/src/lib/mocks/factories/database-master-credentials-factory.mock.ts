import { Chance } from 'chance'
import { Credentials } from 'qovery-typescript-axios'

const chance = new Chance()

export const databaseMasterCredentialsFactoryMock = (howMany: number): Credentials[] =>
  Array.from({ length: howMany }).map((_, index) => ({
    login: chance.name(),
    password: chance.guid(),
    host: chance.name(),
    port: chance.d4(),
  }))
