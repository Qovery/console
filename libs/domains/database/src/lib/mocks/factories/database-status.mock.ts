import { Chance } from 'chance'
import { Status, ServiceDeploymentStatusEnum, StateEnum } from 'qovery-typescript-axios'

const chance = new Chance()

export const databaseStatusFactoryMock = (howMany: number): Status[] =>
  Array.from({ length: howMany }).map((_, index) => ({
    id: `${index}`,
    message: chance.sentence(),
    service_deployment_status: chance.pickone(
      Object.values([
        ServiceDeploymentStatusEnum.UP_TO_DATE,
        ServiceDeploymentStatusEnum.UP_TO_DATE,
        ServiceDeploymentStatusEnum.UP_TO_DATE,
      ])
    ),
    state: chance.pickone(Object.values([StateEnum.DEPLOYED, StateEnum.RUNNING, StateEnum.STOP_ERROR])),
  }))
