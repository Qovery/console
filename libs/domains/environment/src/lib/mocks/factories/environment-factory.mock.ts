import { Chance } from 'chance'
import {
  DeploymentHistoryStatusEnum,
  EnvironmentModeEnum,
  ServiceDeploymentStatusEnum,
  StateEnum,
  WeekdayEnum,
} from 'qovery-typescript-axios'
import { EnvironmentEntity } from '@qovery/shared/interfaces'

const chance = new Chance()

const service = {
  id: chance.integer().toString(),
  created_at: chance.date().toString(),
  updated_at: chance.date().toString(),
  name: chance.name(),
  status: chance.pickone(Object.values([StateEnum.DEPLOYED, StateEnum.RUNNING, StateEnum.STOP_ERROR])),
}

const database = {
  id: chance.integer().toString(),
  created_at: chance.date().toString(),
  updated_at: chance.date().toString(),
  name: chance.name(),
  status: chance.pickone(Object.values([DeploymentHistoryStatusEnum.SUCCESS, DeploymentHistoryStatusEnum.FAILED])),
}

const deployment = {
  id: chance.integer().toString(),
  created_at: chance.date().toString(),
  updated_at: chance.date().toString(),
  status: StateEnum.DEPLOYED,
  applications: [service, service, service],
  databases: [database, database, database],
}

const deploymentRules = {
  id: chance.guid(),
  created_at: chance.date().toString(),
  updated_at: chance.date().toString(),
  auto_deploy: true,
  auto_stop: true,
  auto_delete: true,
  auto_preview: true,
  timezone: 'UTC',
  start_time: chance.date().toString(),
  stop_time: chance.date().toString(),
  weekdays: [chance.pickone(Object.values(WeekdayEnum))],
}

export const environmentFactoryMock = (howMany: number, noStatus = false, noDeployments = false): EnvironmentEntity[] =>
  Array.from({ length: howMany }).map((_, index) => ({
    id: `${(index + 1) * Math.random()}`,
    created_at: chance.date().toString(),
    updated_at: chance.date().toString(),
    name: chance.name(),
    last_updated_by: chance.date().toString(),
    cloud_provider: {
      provider: chance.name(),
      cluster: chance.name(),
    },
    mode: chance.pickone(
      Object.values([EnvironmentModeEnum.DEVELOPMENT, EnvironmentModeEnum.PRODUCTION, EnvironmentModeEnum.STAGING])
    ),
    cluster_id: chance.guid(),
    status: !noStatus
      ? {
          id: `${index}`,
          state: chance.pickone(Object.values([StateEnum.DEPLOYED, StateEnum.RUNNING, StateEnum.STOP_ERROR])),
          message: chance.word({ length: 10 }),
          service_deployment_status: chance.pickone(Object.values([ServiceDeploymentStatusEnum.UP_TO_DATE])),
        }
      : undefined,
    deployments: !noDeployments ? [deployment, deployment, deployment] : undefined,
    deploymentRules: deploymentRules,
  }))
