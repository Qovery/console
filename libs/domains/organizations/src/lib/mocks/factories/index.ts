export * from './organizations-factory.mock'
