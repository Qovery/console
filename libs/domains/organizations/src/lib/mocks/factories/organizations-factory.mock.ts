import { OrganizationInterface } from '../../interfaces/organizations.interface'
import { OrganizationPlanType } from '../../enums/organizations.enum'
import { Chance } from 'chance'

const chance = new Chance()

export const organizationsFactoryMock = (howMany: number): OrganizationInterface[] =>
  Array.from({ length: howMany }).map((_, index) => ({
    id: `${index}`,
    created_at: new Date(),
    updated_at: new Date(),
    name: chance.name(),
    description: chance.word({ length: 10 }),
    plan: chance.pickone(Object.values(OrganizationPlanType)),
    website_url: chance.url(),
    repository: chance.name(),
    logo_url: chance.url(),
    icon_url: chance.url(),
    owner: chance.name(),
  }))
