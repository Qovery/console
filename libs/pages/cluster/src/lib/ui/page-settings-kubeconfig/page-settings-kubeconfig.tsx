import { type Cluster } from 'qovery-typescript-axios'
import { useCallback } from 'react'
import { useDropzone } from 'react-dropzone'
import { SettingsHeading } from '@qovery/shared/console-shared'
import { KUBECONFIG } from '@qovery/shared/routes'
import { Button, Icon, Link, Section } from '@qovery/shared/ui'

export interface PageSettingsKubeconfigProps {
  cluster: Cluster
  onSubmit: (cluster: Cluster, kubeconfig: string) => void
}

export function PageSettingsKubeconfig({ cluster, onSubmit }: PageSettingsKubeconfigProps) {
  const onDrop = useCallback((acceptedFiles: File[]) => {
    acceptedFiles.forEach((file) => {
      const reader = new FileReader()

      reader.onabort = () => console.log('file reading was aborted')
      reader.onerror = () => console.log('file reading has failed')
      reader.onload = () => {
        onSubmit(cluster, reader.result as string)
      }
      reader.readAsText(file)
    })
  }, [])
  const { getRootProps, getInputProps } = useDropzone({
    onDrop,
    maxFiles: 1,
  })

  const searchParams = new URLSearchParams({
    organizationId: cluster.organization.id,
    clusterId: cluster.id,
  })

  return (
    <div className="flex w-full flex-col justify-between">
      <Section className="max-w-content-with-navigation-left p-8">
        <SettingsHeading title="Kubeconfig" description="Upload your Kubeconfig file here." />
        <div className="flex items-center gap-2 rounded border border-neutral-200 p-3">
          <div className="p-2">
            <Icon iconName="file-lines" />
          </div>
          <div className="flex grow flex-col text-xs">
            <span className="font-medium text-neutral-400">Kubeconfig</span>
            <Link
              className="text-xs"
              to={{
                pathname: KUBECONFIG,
                search: searchParams.toString(),
              }}
              target="_blank"
              rel="noopener noreferrer"
            >
              View <Icon iconName="eye" />
            </Link>
          </div>
          <div className="flex gap-2">
            <Button type="button" variant="outline" {...getRootProps()} color="neutral">
              <input {...getInputProps()} />
              <Icon iconName="arrows-rotate" />
            </Button>
          </div>
        </div>
      </Section>
    </div>
  )
}

export default PageSettingsKubeconfig
