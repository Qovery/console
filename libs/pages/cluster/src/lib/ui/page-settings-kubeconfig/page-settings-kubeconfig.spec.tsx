import { clusterFactoryMock } from '@qovery/shared/factories'
import { renderWithProviders } from '@qovery/shared/util-tests'
import { PageSettingsKubeconfig, type PageSettingsKubeconfigProps } from './page-settings-kubeconfig'

const mockCluster = clusterFactoryMock(1)[0]

describe('PageSettingsKubeconfig', () => {
  const props: PageSettingsKubeconfigProps = {
    cluster: mockCluster,
    onSubmit: jest.fn(),
  }

  it('should render successfully', async () => {
    const { baseElement } = renderWithProviders(<PageSettingsKubeconfig {...props} />)
    expect(baseElement).toBeTruthy()
  })

  it('should match snapshot', async () => {
    const { container } = renderWithProviders(<PageSettingsKubeconfig {...props} />)
    expect(container).toMatchSnapshot()
  })
})
