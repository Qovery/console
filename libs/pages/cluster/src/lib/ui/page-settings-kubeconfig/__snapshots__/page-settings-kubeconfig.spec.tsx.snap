// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageSettingsKubeconfig should match snapshot 1`] = `
<div>
  <div
    class="flex w-full flex-col justify-between"
  >
    <section
      class="flex flex-col max-w-content-with-navigation-left p-8"
    >
      <div
        class="mb-8 flex w-full justify-between gap-2"
      >
        <div>
          <h1
            class="text-neutral-400 text-xl font-bold mb-2"
          >
            Kubeconfig
          </h1>
          <p
            class="mb-3 max-w-lg text-sm text-neutral-400"
          >
            Upload your Kubeconfig file here.
          </p>
          <button
            class="inline-flex shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 outline-0 shadow-sm disabled:shadow-none active:shadow-none outline-neutral-500 text-xs h-6 px-2 rounded bg-brand-50 border border-brand-500 text-brand-500 hover:[&:not(:active):not(:disabled)]:bg-white w-fit items-center gap-1"
          >
            <i
              aria-hidden="true"
              class="fa-regular fa-circle-question p-0.5 text-xs"
            />
            <span>
              Need help here?
            </span>
          </button>
        </div>
      </div>
      <div
        class="flex items-center gap-2 rounded border border-neutral-200 p-3"
      >
        <div
          class="p-2"
        >
          <i
            aria-hidden="true"
            class="fa-solid fa-file-lines "
          />
        </div>
        <div
          class="flex grow flex-col text-xs"
        >
          <span
            class="font-medium text-neutral-400"
          >
            Kubeconfig
          </span>
          <a
            class="cursor-pointer transition duration-100 font-medium inline-flex items-center gap-1 text-sky-500 hover:text-sky-600 text-xs"
            href="/kubeconfig?organizationId=123&clusterId=0"
            rel="noopener noreferrer"
            target="_blank"
          >
            View 
            <i
              aria-hidden="true"
              class="fa-solid fa-eye "
            />
          </a>
        </div>
        <div
          class="flex gap-2"
        >
          <button
            class="inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 outline-0 shadow-sm disabled:shadow-none active:shadow-none bg-white dark:bg-neutral-600 outline-brand-500 text-xs h-7 px-2 rounded border border-neutral-250 dark:border-neutral-400 text-neutral-400 dark:text-white hover:[&:not(:active)]:border-neutral-300 dark:hover:[&:not(:active):not(:disabled)]:border-neutral-350 active:bg-neutral-150 disabled:text-neutral-300 disabled:bg-neutral-100 disabled:border-none"
            role="presentation"
            tabindex="0"
            type="button"
          >
            <input
              multiple=""
              style="display: none;"
              tabindex="-1"
              type="file"
            />
            <i
              aria-hidden="true"
              class="fa-solid fa-arrows-rotate "
            />
          </button>
        </div>
      </div>
    </section>
  </div>
</div>
`;
